presto-parser/src/main/java/com/facebook/presto/sql/tree/PrincipalSpecification.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import java.util.Objects;
+
+import static com.facebook.presto.sql.tree.PrincipalSpecification.Type.CURRENT_ROLE;
+import static com.facebook.presto.sql.tree.PrincipalSpecification.Type.CURRENT_USER;
+import static com.facebook.presto.sql.tree.PrincipalSpecification.Type.ROLE;
+import static com.facebook.presto.sql.tree.PrincipalSpecification.Type.UNSPECIFIED;
+import static com.facebook.presto.sql.tree.PrincipalSpecification.Type.USER;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class PrincipalSpecification
+{
+    public enum Type
+    {
+        UNSPECIFIED, USER, ROLE, CURRENT_USER, CURRENT_ROLE
+    }
+
+    private final Type type;
+    private final String name;
```

Comment:
I've started to implement this, but it just introduces additional code with no value. This field is completely private. We never expose that null.